### PR TITLE
Use WaitForExit(timeout) rather than loop on HasExited property for some tests

### PR DESCRIPTION
**Pull request by KevinRansom**
For an age now the ci_part2 has hung perhaps one time in 3 or 4.  This is super irritating and a waste of CI resources.

I was about to remove what I think the offending tests are, but I noticed that they loop on the process.HasExited property.

I can easily imagine since the symptom is a hang that the process class may not be super good about setting this property.  So instead I changed it to the WaitForExit event with a time out of 30 seconds.

Hopefully this will cause the tests to not hang anymore, and behave properly.

### CODE REVIEW

**Comment by KevinRansom** [comment]
Last one worked:
@dotnet-bot test Windows_NT Release_ci_part2 Build please

File: build.cmd [hunk 1]
```diff
@@ -1,4 +1,4 @@
-rem Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+ rem Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
```
Comment (brettfo) [inline build.cmd]: Nit: undo this

File: tests/fsharpqa/Source/CodeGen/Structure/DoNotInlineX.fs [hunk 2]
```diff
@@ -4,6 +4,8 @@ open System
 open System.Diagnostics
 open System.IO
 
+let longtime = 30000                    // longtime is 30 seconds
```
Comment (brettfo) [inline tests/fsharpqa/Source/CodeGen/Structure/DoNotInlineX.fs]: To make this more explicit I'd prefer `TimeSpan.FromSeconds(30)` and `.WaitForExit(longtime.TotalMilliseconds)`.  Magic pre-computed numbers scare me.

File: tests/service/FscTests.fs [hunk 3]
```diff
@@ -76,6 +76,8 @@ type PEVerifier () =
 #endif
 
     static let execute (fileName : string, arguments : string) =
+        // Peverify may run quite a while some assemblies are pretty big.  Make the timeout 3 minutes just in case.
+        let longtime = 180000
```
Comment (brettfo) [inline tests/service/FscTests.fs]: Same `TimeSpan` comment as above.

File: build.cmd (hunk 1 above)
Comment (KevinRansom) [reply build.cmd]: oh yeah the bump space ...

**Comment by KevinRansom** [comment]
@vasily-kirichenko Good grief ... was that API designed or coughed up in an influluenza season

**Comment by KevinRansom** [comment]
@vasily-kirichenko Hmmm I might have got a trifle emotional there.  Oops!!!

**Comment by vasily-kirichenko** [comment]
@KevinRansom :) The API is not good indeed.

**Pull request merged**